compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaRecordTypeSymbol.java

```diff
@@ -103,11 +121,44 @@ public BallerinaRecordTypeSymbol(CompilerContext context, ModuleID moduleID, BRe
         return this.typeInclusions;
     }
 
+    public List<TypeSymbol> originalTypeInclusions() {
+        if (this.originalTypeInclusions == null) {
+            TypesFactory typesFactory = TypesFactory.getInstance(this.context);
+            List<BType> inclusions = ((BRecordType) this.getBType()).typeInclusions;
+
+            List<TypeSymbol> typeRefs = new ArrayList<>();
+            for (BType inclusion : inclusions) {
+                if (inclusion.tag != TypeTags.NONE
+                        && !(inclusion.tsymbol.pkgID.getOrgName().getValue().equals("ballerina")
+                        && inclusion.tsymbol.pkgID.getNameComps().size() > 0
+                        && inclusion.tsymbol.pkgID.getNameComps().get(0).getValue().equals("lang"))) {
+
+                    TypeSymbol type = typesFactory.getTypeDescriptor(inclusion);
+
+                    // If the inclusion was not a type ref, the type would be semantic error and the type factory will
+                    // return null. Therefore, skipping them.
+                    if (type != null) {
+                        typeRefs.add(type);
+                    }
+                }
+            }
+
+            this.originalTypeInclusions = Collections.unmodifiableList(typeRefs);
+        }
+
+        return this.originalTypeInclusions;
+    }
+
     @Override
     public String signature() {
         // Treating every record typedesc as exclusive record typedescs.
         StringJoiner joiner = new StringJoiner(" ", "{| ", " |}");
-        for (RecordFieldSymbol fieldSymbol : this.fieldDescriptors().values()) {
+        for (TypeSymbol typeInclusion : this.originalTypeInclusions()) {
+            String ballerinaTypeSignature = "*" + typeInclusion.signature() + ";";
```

Comment:
Misleading name. Not a type signature. Alternatively can just simply use a chain of `add()` calls.